internal/gengapic/genrest.go

```diff
@@ -144,23 +143,152 @@ func (g *generator) genRESTMethod(servName string, serv *descriptor.ServiceDescr
 	if pf, err := g.pagingField(m); err != nil {
 		return err
 	} else if pf != nil {
-		// TODO(dovs)
-		return nil
+		iter, err := g.iterTypeOf(pf)
+		if err != nil {
+			return err
+		}
+
+		return g.pagingRESTCall(servName, m, pf, iter)
 	}
 
 	switch {
 	case m.GetClientStreaming():
-		// TODO(dovs)
-		return nil
+		return g.noRequestStreamRESTCall(servName, serv, m)
 	case m.GetServerStreaming():
-		// TODO(dovs)
-		return nil
+		return g.serverStreamRESTCall(servName, serv, m)
 	default:
-		// TODO(dovs)
 		return g.unaryRESTCall(servName, m)
 	}
 }
 
+func (g *generator) serverStreamRESTCall(servName string, s *descriptor.ServiceDescriptorProto, m *descriptor.MethodDescriptorProto) error {
+	// Streaming calls will most likely NEVER be supported for REST clients,
+	// but the interface signature must be preserved.
+	// Making sure not to call streaming methods on a REST client, or checking for
+	// errors in a situation with mixed gRPC and REST clients, is left to user code.
```

Comment:
I think we should error eagerly when calling an unimplemented REST method and minimize the burden on the developer. Sounds to me like that's what you're doing below, so your comment is slightly confusing. The user is already checking for errors. Maybe simplify to ~"Users should not call unimplemented REST methods, which will error"?

Reply:
Changed wording to say "Unimplemented REST methods will always error."